File: Design Patterns/basic-patterns-java/decorator/ConcreteCircle.java
package decorator;

import javafx.scene.paint.Color;
import javafx.scene.shape.Circle;

public class ConcreteCircle extends Circle implements Component {

    public ConcreteCircle(double radius) {
        super(radius);
    }

    @Override
    public void operation() {
        this.setFill(Color.TRANSPARENT);
        this.setStroke(Color.BLACK);
    }

    @Override
    public Circle getCircle() {
        return this;
    }

}

File: Design Patterns/basic-patterns-java-kotlin/decorator/ConcreteCircle.kt
package decorator

import javafx.scene.paint.Color
import javafx.scene.shape.Circle

class ConcreteCircle(radius: Double) : Circle(radius), Component {
    override fun operation() {
        fill = Color.TRANSPARENT
        stroke = Color.BLACK
    }

    override val circle: Circle
        get() = this
}